compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -1053,6 +1053,12 @@ where
                 })
             }
 
+            ty::Adt(adt_def, ..) if adt_def.is_maybe_dangling() => {
+                // FIXME: what is the exact effect of maybe dangling?
+                Self::ty_and_layout_pointee_info_at(this.field(cx, 0), cx, offset)
+                    .map(|info| PointeeInfo { safe: None, ..info })
+            }
```

Comment:
This should also reset the `size` to 0, that's how we encode `dereferenceable`.

Odd that the test does not catch this...